### PR TITLE
added @mention feature

### PR DESCRIPTION
**Pull request by YKuvonchbek**
It works for group topics like Whatsapp

### CODE REVIEW

**Comment by Kuvonchbeky** [comment]
It works only group topics like Whatsapp

File: app/src/main/java/co/tinode/tindroid/MessagesFragment.java [hunk 1]
```diff
@@ -274,20 +372,13 @@ public void onResume() {
...
-            mTopicName = null;
-        }
-
         if (mTopicName != null) {
             mTopic = (ComTopic<VxCard>) Cache.getTinode().getTopic(mTopicName);
             runMessagesLoader(mTopicName);
         } else {
             mTopic = null;
         }
+        if (mTopic != null && mTopic.isGrpType()) notifyDataSetChanged();
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/MessagesFragment.java]: Why do you need to reset it even if there are no mentions?

File: app/src/main/java/co/tinode/tindroid/MessagesFragment.java [hunk 2]
```diff
@@ -677,13 +768,38 @@ private void sendText() {
         if (activity == null) {
             return;
         }
+        List<Drafty.Entity> exEnt = null;
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/MessagesFragment.java]: The original code keeps conversion of text into `Drafty` all in one place: in `Drafty.parse(String text)`. What you are doing here is creating another parser just for the mentions. I don't think it's right. You should keep all parsing in one place. If you need to pass some data to Drafty parser, pass a callback.

File: app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java [hunk 3]
```diff
@@ -271,7 +273,21 @@ public void onClick(View widget) {
...
                     } catch (ClassCastException | NullPointerException ignored) {}
                     break;
-                case "MN": break;
+                case "MN":
+                    span = new TreeNode(new MentionSpan("") {
+                        @Override
+                        public void onClick(View widget) {
+                            String checkMention = data.get("val").toString();
+                            if (checkMention.charAt(0) == '@') {
+                                Toast.makeText(mContainer.getContext(), "(@)This mentions aren't supported yet", Toast.LENGTH_SHORT).show();
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java]: All literal text should be moved to resource file.

Comment (or-else) [reply app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java]: I don't think formatter should show toasts.

File: app/src/main/java/co/tinode/tindroid/MessagesFragment.java [hunk 4]
```diff
@@ -1128,4 +1243,289 @@ public void run() {
...
+            final View.OnClickListener action = new View.OnClickListener() {
+                @SuppressLint("SetTextI18n")
+                @Override
+                public void onClick(View v) {
+                    int position = holder.getAdapterPosition();
+                    final Subscription<VxCard, PrivateType> sub = mItemsFiltered.get(position);
+                    VxCard pub = mTopic.getPub();
+                    String text = editor.getText().toString().substring(1);
+                    editor.setText("@" + text.replace(text, sub.user));
+//                    showMemberAction(pub != null ? pub.fn : null, holder.name.getText().toString(), sub.user,
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/MessagesFragment.java]: Please remove all commented-out and otherwise unused code out of the PR.

**Comment by or-else** [comment]
Overall this PR seems to be dependent on https://github.com/tinode/tindroid/pull/73

Please follow these rules:

One feature - one PR. 

**Not** two features - one PR. **Not** one feature - two PRs.

One feature - one PR. 

Thanks!

**Pull request closed without merging**